src/internal/file_operatio_compress_test.go

```
@@ -0,0 +1,315 @@
+	testFiles := map[string]string{
+
+		filepath.Join(testDir1, "file1.txt"): "Content of file1",
+
+		filepath.Join(subDir, "file2.txt"): "Content of file2",
+
+		filepath.Join(testDir2, "file3.txt"): "Content of file3",
+	}
+
+	for filePath, content := range testFiles {
+
+		err := os.WriteFile(filePath, []byte(content), 0644)
```

golangci-lint (1.64.8)

shadow: declaration of "err" shadows declaration at line 27

(govet)